fabric-api-base/src/main/java/net/fabricmc/fabric/api/util/TriMap.java

```diff
@@ -0,0 +1,52 @@
+package net.fabricmc.fabric.api.util;
+
+import java.util.Objects;
+import java.util.Set;
+
+import com.google.common.collect.Sets;
+
+public class TriMap<K, L, R> {
+
+	private final Set<Entry<K, L, R>> entries = Sets.newHashSet();
+
+	private void addEntry(K key, L left, R right) {
+		this.entries.add(new Entry<>(key, left, right));
+	}
+
+	public void add(K key, L left, R right) {
+		for(Entry<K, L, R> entry : entries){
```

Comment:
This is a set, use just use entries.add

Reply:
It actually just checks if it exists or not, and then it adds it

Reply:
this is a set my man, it doesn't allow duplicates by the hash

Reply:
Done.